src/test/groovy/com/github/rodionovsasha/commentservice/services/UserServiceTest.groovy

```diff
@@ -0,0 +1,184 @@
+import com.github.rodionovsasha.commentservice.exceptions.UserNotFoundException
+import com.github.rodionovsasha.commentservice.repositories.UserRepository
+import org.springframework.beans.factory.annotation.Autowired
+
+class UserServiceTest extends BaseTest {
+    @Autowired
+    UserService userService
+    @Autowired
+    UserRepository userRepository
+
+    def "getActiveUser returns active user by id"() {
+        when:
+        def user = userService.getActiveUser(1)
+
+        then:
+        user.name == "Homer"
+        user.age == 39
+        user.enabled
+    }
+
+    def "getActiveUser throws when user is inactive"() {
+        when:
+        userService.getActiveUser(2)
+
+        then:
+        def e = thrown(InactiveUserException)
+        e.message == "The user with id '2' is not active"
+    }
+
+    def "getActiveUser throws when user not found"() {
+        when:
+        userService.getActiveUser(999)
+
+        then:
+        def e = thrown(UserNotFoundException)
+        e.message == "The user with id '999' could not be found"
+    }
+
+    def "updateName does update user's name"() {
+        given:
+        userRepository.getOne(1L).name == "Homer"
+
+        when:
+        userService.updateName(1, "Maggie")
+        and:
+        def user = userRepository.getOne(1L)
+
+        then:
+        user.name == "Maggie"
+    }
+
+    def "updateName does not change anything except name"() {
+        when:
+        userService.updateName(1, "Maggie")
+        and:
+        def user = userRepository.getOne(1L)
+
+        then:
+        user.age == 39
+        user.enabled
+    }
+
+    def "updateName throws when user is not active"() {
+        when:
+        userService.updateName(2, "Maggie")
+
+        then:
+        def e = thrown(InactiveUserException)
+        e.message == "The user with id '2' is not active"
+    }
+
+    def "updateAge does update user's age"() {
+        given:
+        userRepository.getOne(1L).age == 39
+
+        when:
+        userService.updateAge(1, 35)
+        and:
+        def user =  userRepository.getOne(1L)
+
+        then:
+        user.age == 35
+    }
+
+    def "updateAge does not change anything except age"() {
+        when:
+        userService.updateAge(1, 35)
+        and:
+        def user = userRepository.getOne(1L)
+
+        then:
+        user.name == "Homer"
+        user.enabled
+    }
+
+    def "updateAge throws when user is not active"() {
+        when:
+        userService.updateAge(2, 35)
+
+        then:
+        def e = thrown(InactiveUserException)
+        e.message == "The user with id '2' is not active"
+    }
+
+    def "deactivate makes user inactive"() {
+        given:
+        userRepository.getOne(1L).enabled
+
+        when:
+        userService.deactivate(1)
+        and:
+        def user =  userRepository.getOne(1L)
+
+        then:
+        !user.enabled
+    }
+
+    def "deactivate does not change anything except state"() {
+        when:
```

Comment:
Let's avoid this cranky `and` after the `when`. It could look easier:
```groovy
when:
userService.deactivate(1)

then:
def user = userRepository.getOne(1L) // Spock basically checks that user is not null on this line
user.name == "Homer"
user.age == 39
```